src/main/java/org/opensearch/knn/plugin/transport/UpdateModelGraveyardTransportAction.java

```diff
@@ -146,6 +161,7 @@ public ClusterTasksResult<UpdateModelGraveyardTask> execute(ClusterState cluster
                     modelGraveyard.remove(task.getModelId());
                     continue;
                 }
+                checkIfIndicesAreUsingModel(clusterState, task);
```

Comment:
Can we make this method return true and then we throw the exception in this method?

Reply:
that sounds good